Sources/SwiftDocC/Model/Rendering/References/ExternalLocationReference.swift

```diff
@@ -0,0 +1,53 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+/// A specialized ``DownloadReference`` used for ``CallToAction`` directives.
+///
+/// `@CallToAction` directives can link either to a local file or to a URL, whether relative or
+/// absolute. Directives that use the `file` argument will create a ``DownloadReference`` and copy
+/// the file from the catalog into the resulting archive.
+///
+/// An `ExternalLocationReference` is intended to encode to Render JSON compatible with a
+/// ``DownloadReference``, but with the `url` set to the text given in the `@CallToAction`'s `url`
+/// argument.
+public struct ExternalLocationReference: RenderReference, URLReference {
+    public static var baseURL: URL = .init(string: "/\(DownloadReference.locationName)/")!
+
+    public var type: RenderReferenceType = .download
+
+    public var identifier: RenderReferenceIdentifier
+
+    enum CodingKeys: String, CodingKey {
+        case type
+        case identifier
+        case url
+    }
+
+    public init(identifier: RenderReferenceIdentifier) {
+        self.identifier = identifier
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        self.identifier = try container.decode(RenderReferenceIdentifier.self, forKey: .identifier)
```

Comment:
We have a separate discussion about the round-tripping already – but shouldn't we be encoding the type here as well?

Reply:
Encode the type how? The idea behind this type is that it should encode to JSON compatible with a `DownloadReference`, including the `type` property being set to `"download"`. Are you suggesting that we add an extra field, not used by Swift-DocC-Render, that would indicate the ultimate source of a `"download"` reference?

Reply:
No – sorry I had a type above. I mean *decode* not *encode*.

We're already encoding the property below, shouldn't we be decoding it here?

```suggestion
        self.identifier = try container.decode(RenderReferenceIdentifier.self, forKey: .identifier)
        self.type = try container.decode(RenderReferenceType.self, forKey: .type)
```